src/tls13.c

```diff
@@ -4100,26 +4144,21 @@ int SendTls13ClientHello(WOLFSSL* ssl)
         args->idx += DTLS_RECORD_EXTRA + DTLS_HANDSHAKE_EXTRA;
 #endif /* WOLFSSL_DTLS13 */
 
-    /* Version | Random | Session Id | Cipher Suites | Compression */
-    args->length = VERSION_SZ + RAN_LEN + ENUM_LEN + suites->suiteSz +
+    /* Version | Random | Cipher Suites | Compression */
+    args->length = VERSION_SZ + RAN_LEN + suites->suiteSz +
             SUITE_LEN + COMP_LEN + ENUM_LEN;
+#if defined(WOLFSSL_TLS13_MIDDLEBOX_COMPAT)
+    ssl->options.tls13MiddleBoxCompat = 1;
```

Comment:
Why are we forcing this here?
It is meant to be something the application chooses to do.

Reply:
That's how it was before my changes. I just preserved the logic with my changes. As to why its forced, I think that when the middlebox macro is on we are supposed to always be acting middlebox compatible.

Reply:
I think the logic was that we are sending the session Id and it is middle box compatibility thing and so we are now doing compatibility.
I don't want it forced on. We have chosen to always send it now regardless of compatibility.

Reply:
This line is only in effect when `WOLFSSL_TLS13_MIDDLEBOX_COMPAT` is defined. This is later overwritten if we are doing quic. This is the same logic as before. Previously, when `WOLFSSL_TLS13_MIDDLEBOX_COMPAT` was defined we would also always force middlebox compatibility. Would you like me to change the behaviour in this PR? I'm not sure if we don't force middlebox compatibility in other locations too. Those would also need to be reviewed.

Reply:
You are refusing to make a requested change.
Fine.
Do another PR with the change and review all other instances.

Reply:
I don't mind making the change. I'm just making sure that we are on the same page about what the requested end result is. I'll tackle this in a new PR.